### PR TITLE
start adding unit tests for Ironic.UpdateHardwareState

### PR DESCRIPTION
**Pull request by andfasano**
Increasing test coverage for Ironic provisioner

### CODE REVIEW

File: pkg/provisioner/ironic/testserver/ironic.go [hunk 1]
```diff
@@ -58,3 +63,21 @@ func (m *IronicMock) WithDrivers() *IronicMock {
 	`)
 	return m
 }
+
+// WithNode configures the server with a valid response for /v1/nodes
+func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
```
Comment (dhellmann) [inline pkg/provisioner/ironic/testserver/ironic.go:68]: Some of these methods are already in master. Maybe this PR needs to be rebased?

File: pkg/provisioner/ironic/updatehardwarestate_test.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			}
+			auth := clients.AuthConfig{Type: clients.NoAuth}
+			prov, err := newProvisionerWithSettings(host, bmc.Credentials{}, publisher,
+				tc.ironic.Endpoint(), auth, tc.inspector.Endpoint(), auth,
+			)
+			if err != nil {
+				t.Fatalf("could not create provisioner: %s", err)
+			}
+
+			prov.status.ID = nodeUUID
```
Comment (dhellmann) [inline pkg/provisioner/ironic/updatehardwarestate_test.go:124]: This should be set on host after it is created on line 109.

**Review by dhellmann — CHANGES_REQUESTED** [review]

File: pkg/provisioner/ironic/inspecthardware_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
+package ironic
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
```
[the remark this answers is not in the document]
Comment (andfasano) [reply pkg/provisioner/ironic/inspecthardware_test.go]: Done

File: pkg/provisioner/ironic/testserver/ironic.go (hunk 1 above)
Comment (andfasano) [reply pkg/provisioner/ironic/testserver/ironic.go]: Done

File: pkg/provisioner/ironic/updatehardwarestate_test.go (hunk 2 above)
Comment (andfasano) [reply pkg/provisioner/ironic/updatehardwarestate_test.go]: Thanks, good catch

**Comment by dhellmann** [comment]
/lgtm

I'll leave it to you to cancel the hold, in case you're still working on something.

**Pull request merged**